cmake/GenerateMetadata.cmake

```diff
@@ -1,8 +1,11 @@
 macro(GenerateMetadata _target)
     add_dependencies(${_target} MetadataGenerator)
     add_custom_command(TARGET ${_target}
-        POST_BUILD
+        PRE_BUILD
```

Comment:
Can this be made it in the `PRE_LINK` phase, so we don't generate metadata before each failed compile